Environment_Engine/Query/InvalidPanels.cs

```diff
@@ -0,0 +1,144 @@
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using BH.oM.Environment;
+using BH.Engine.Geometry;
+using BH.oM.Environment.Elements;
+
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+
+namespace BH.Engine.Environment
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Return a collection of panels which we identified Invalid data 1-external adj space, 2-shade check, 3-internal adj space, 4-internal two diff spaces, 5-area to small, 6-segment to short, 7-polyline closed, 8-poltyline interesect")]
+        [Input("panels", "A collection of environment panels")]
+        [Input("panelArea", "A min panel area to return warning")]
+        [Input("panelEdgeLength", "A min panelEdgeLength to return warning")]
+        [Output("invalidPanels", "A collection of panels which are invalid group in sublist for each of 8 checks that we perform, see description")]
+        public static List<List<Panel>> InvalidPanels(this List<Panel> panels, double panelArea = 0.15, double panelEdgeLength = 0.15)
+        {
+            List<List<Panel>> rtn = new List<List<Panel>>();
+
+            for (int i = 0; i < 8; i++)
+                rtn.Add(new List<Panel>());
+            foreach (Panel p in panels)
+
+            {
+                //check for external
+                if ((p.Type == PanelType.Roof || p.Type == PanelType.WallExternal || p.Type == PanelType.FloorExposed || p.Type == PanelType.FloorRaised || p.Type == PanelType.SlabOnGrade || p.Type == PanelType.UndergroundSlab || p.Type == PanelType.UndergroundWall) && p.ConnectedSpaces.Count != 1)
+                {
+                    
+                    rtn[0].Add(p);
+                }
+
+                //check for shade
+                if ((p.Type == PanelType.Shade || p.Type == PanelType.SolarPanel) && p.ConnectedSpaces.Count != 0)
+                {
+
+                    rtn[1].Add(p);
+                }
+
+                //check for internal
+                if ((p.Type == PanelType.Ceiling || p.Type == PanelType.WallInternal || p.Type == PanelType.FloorInternal || p.Type == PanelType.UndergroundCeiling) && p.ConnectedSpaces.Count != 2 )
+                {
+                   
+
+                    rtn[2].Add(p);
+
+                }
+
+                //check for internal to make sure we have two different space
+                if ( p.ConnectedSpaces.Count == 2 && p.ConnectedSpaces[0] == p.ConnectedSpaces[1])
+                {
+
+
+                    rtn[3].Add(p);
+
+                }
+
+                //check if geometry area is too small
+                if (p.Area() < panelArea)
+                {
+
+                    rtn[4].Add(p);
+                }
+                else
```

Comment:
Why is this else statement here? Why are we only checking edge length if the area is bigger than tolerance? Why can't we check this as well?